sdk/include/opentelemetry/sdk/trace/tracer_provider.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "opentelemetry/sdk/trace/tracer.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+using opentelemetry::nostd::string_view;
+using opentelemetry::sdk::trace::Tracer;
+
+namespace opentelemetry
+{
+namespace sdk
+{
+namespace trace
+{
+class TracerProvider : public opentelemetry::trace::TracerProvider
+{
+public:
+  Tracer *const GetTracer(string_view library_name, string_view library_version = "") override;
+
+private:
+  std::vector<std::unique_ptr<Tracer>> tracers_;
```

Comment:
Use an std::shared_ptr for Tracer so that we can use [std::enable_shared_from_this](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this)